hive-metastore/src/main/java/org/apache/iceberg/hive/legacy/LegacyHiveTableUtils.java

```diff
@@ -56,7 +56,26 @@ static Schema getSchema(org.apache.hadoop.hive.metastore.api.Table table) {
     String schemaStr = props.get("avro.schema.literal");
     Schema schema;
     if (schemaStr != null) {
-      schema = AvroSchemaUtil.toIceberg(new org.apache.avro.Schema.Parser().parse(schemaStr));
+      org.apache.avro.Schema avroSchema = new org.apache.avro.Schema.Parser().parse(schemaStr);
+      org.apache.avro.Schema hiveSchema = LegacyHiveSchemaUtils.convertHiveSchemaToAvro(table);
+      org.apache.avro.Schema avroSchemaWithoutNullable =
+          LegacyHiveSchemaUtils.extractActualTypeIfFieldIsNullableTypeRecord(avroSchema);
+      org.apache.avro.Schema hiveSchemaWithoutNullable =
+          LegacyHiveSchemaUtils.extractActualTypeIfFieldIsNullableTypeRecord(hiveSchema);
+
+      org.apache.avro.Schema tableSchema = avroSchema;
+      boolean isHiveSchemaEvolved =
+          LegacyHiveSchemaUtils.isRecordSchemaEvolved(avroSchemaWithoutNullable, hiveSchemaWithoutNullable);
+
+      if (isHiveSchemaEvolved) {
```

Comment:
I am not sure why this check is required. Could you explain why it matters? If it turns it is not needed, this patch will be significantly simplified.